packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeFakeSignature.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  AccessListEIP2930Transaction,
+  FeeMarketEIP1559Transaction,
+  TxData,
+} from "@nomicfoundation/ethereumjs-tx";
+import { Address } from "@nomicfoundation/ethereumjs-util";
+
+import { createNonCryptographicHashBasedIdentifier } from "../../../util/hash";
+
+// Produces a signature with r and s values taken from a hash of the inputs.
+export function makeFakeSignature(
+  tx: TxData | AccessListEIP2930Transaction | FeeMarketEIP1559Transaction,
+  sender: Address
+): {
+  r: number;
+  s: number;
+} {
+  const hashInputString = [
+    sender,
+    tx.nonce,
+    tx.gasLimit,
+    tx.value,
+    tx.to,
+    tx.data,
+    "gasPrice" in tx ? tx.gasPrice : "",
+    "chainId" in tx ? tx.chainId : "",
+    "maxPriorityFeePerGas" in tx ? tx.maxPriorityFeePerGas : "",
+    "maxFeePerGas" in tx ? tx.maxFeePerGas : "",
+    "accessList" in tx
+      ? tx.accessList?.map(([buf, bufs]) =>
+          [buf, ...bufs].map((b) => b.toString()).join(";")
```

Comment:
```suggestion
          [buf, ...bufs].map((b) => b.toString("hex")).join(";")
```

Otherwise you get things like this:

```
> Buffer.from([220, 221]).toString()
'��'
```

Reply:
Great call, thank you. Note that we're also getting something similar for `tx.data` (similar to `'��'`); however, I tried adding `?.toString("hex")::""` to the end of `tx.data`, but it seems that `BufferLike` doesn't accept that arg the way `Buffer` does.

Reply:
I think we should avoid the `.map((a) => a?.toString() ?? "")`. I would try doing something like this:

```js
const hashInputs: Buffer[] = [
  // buffers
]

const hashInputsString = hashInputs
  .map(x => x.toString("hex"))
  .join(",")
```

This means not using a different separator for access lists... but that's probably fine.